src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -217,33 +239,65 @@ private void ReportDiskUsage()
             {
                 if (Directory.Exists(directory))
                 {
+                    bool tryAlternateDelete = false;
                     Console.Write($"Attempting to cleanup {directory} ... ");
-                    Directory.Delete(directory, true);
-                    Console.WriteLine("Success");
+                    try
+                    {
+                        Directory.Delete(directory, true);
+                        Log.LogMessage("Success");
+                    }
+                    catch (PathTooLongException)
+                    {
+                        tryAlternateDelete = true;
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        // This may be being deleted by another thread, OR be > 260 chars
+                        tryAlternateDelete = true;
+                    }
+                    if (tryAlternateDelete && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        Log.LogMessage($"Attempting to work around PathTooLongException by using robocopy and windows shell to delete {directory} ");
+                        Process.Start("robocopy.exe", $"/mir {GetUniqueEmptyFolder()} {directory}").WaitForExit();
```

Comment:
Done, and added some parameters to Robocopy to make its spew less verbose, but still include file paths cleaned up in logs.